main_test.go

```diff
@@ -95,3 +95,60 @@ func TestCopyFromCommandLineArgument(t *testing.T) {
 		t.Errorf("Expected %s but got %s", content, copied)
 	}
 }
+
+func TestParseContentDirectText(t *testing.T) {
+	directText := "Direct text input"
+	contentStr, err := parseContent(&directText, []string{})
+	if err != nil {
+		t.Errorf("Expected no error, got %v", err)
+	}
+	if contentStr != directText {
+		t.Errorf("Expected contentStr to be '%s', got '%s'", directText, contentStr)
+	}
+}
+
+func TestParseContentStdin(t *testing.T) {
+	stdinInput := "Stdin input\n"
+	r, w, _ := os.Pipe()
```

Comment:
The very same way you created createTempFile, could do this

```suggestion

function replaceArgs(t *testing.T) (r *os.File, w *os.File) 
	       r, w, err := os.Pipe()
         if err != nil {
              t.Fail(err)
         }

         originalStdin := os.Stdin 
         t.Cleanup (func() { 
                 os.Stdin = originalStdin 
                 r.Close() 
                 w.Close() 
         }) 
         os.Stdin = r
```

This way you can reuse it safely everywhere

Reply:
I asked for code review on slack.

Someone suggested using testscript

Someone made a proof of concept.

It allows to do not overload os.Std*

https://github.com/datacharmer/clipper/blob/add-testscript/cli%2Fclipper_test.go

Reply:
Thank you @datacharmer, BTW